docker-java/src/test/java/com/github/dockerjava/cmd/EventsCmdIT.java

```diff
@@ -113,6 +114,57 @@ public void testEventStreamingWithFilter() throws Exception {
         }
     }
 
+    @Test
+    public void testEventStreamingWithContainerEventTypeFilter() throws Exception {
+        assumeNotSwarm("", dockerRule);
+
+        String startTime = getEpochTime();
+        int expectedEvents = 1;
+
+        EventsTestCallback eventCallback = new EventsTestCallback(expectedEvents);
+
+        dockerRule.getClient().eventsCmd()
+            .withSince(startTime)
+            .withEventTypeFilter(EventType.CONTAINER)
+            .withEventFilter("start")
+            .exec(eventCallback);
+
+        generateEvents();
+
+        List<Event> events = eventCallback.awaitExpectedEvents(30, TimeUnit.SECONDS);
+
+        // we should only get "start" events here
+        for (Event event : events) {
+            assertThat("Received event: " + event, event.getAction(), is("start"));
+            assertThat("Received event: " + event, event.getType(), is(EventType.CONTAINER));
+        }
+    }
+
+    @Test
+    public void testEventStreamingWithImageEventTypeFilter() throws Exception {
+        assumeNotSwarm("", dockerRule);
+
+        String startTime = getEpochTime();
+        int expectedEvents = 1;
+
+        EventsTestCallback eventCallback = new EventsTestCallback(expectedEvents);
+
+        dockerRule.getClient().eventsCmd()
+            .withSince(startTime)
+            .withEventTypeFilter(EventType.IMAGE)
+            .exec(eventCallback);
+
+        generateEvents();
+
+        List<Event> events = eventCallback.awaitExpectedEvents(30, TimeUnit.SECONDS);
+
+        // we should only get "pull" events here
+        for (Event event : events) {
+            assertThat("Received event: " + event, event.getAction(), is("pull"));
```

Comment:
unrelated to this test:
```suggestion
        for (Event event : events) {
```

Reply:
The test is for IMAGE so it is related.

Reply:
You're testing the event type filtering, not action filtering. If the action filtering is broken, we don't want this test to fail as well and give a misleading result